### PR TITLE
added linux, macos, and common keys for metric providers; gmt now che…

### PR DESCRIPTION
**Pull request by dan-mm**
…cks system architecture and only loads the appropriate metric provider; added get_achitecture to utils

### CODE REVIEW

File: metric_providers/lm_sensors/abstract_provider.py [hunk 1]
```diff
@@ -3,15 +3,19 @@
...
+current_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(f"{current_dir}/../../lib")
+import utils
 from metric_providers.base import BaseMetricProvider
 
 class LmSensorsProvider(BaseMetricProvider):
 
     def _create_options(self):
-        provider_config = GlobalConfig().config['measurement']['metric-providers'][self._provider_config_path]
+        provider_config = GlobalConfig().config['measurement']['metric-providers'][utils.get_architecture()]\
```
Comment (ArneTR) [inline metric_providers/lm_sensors/abstract_provider.py:17]: I changed this in the PR to have the static key "linux". I think it makes no sense to have this provider architecture agnostic, when it is only available on linux anyway (see: https://formulae.brew.sh/formula/lm-sensors)

**Review by ArneTR — APPROVED** [review]
nice work, tests all work for me. Merged

File: lib/utils.py [hunk 2]
```diff
@@ -29,5 +31,20 @@ def get_pascal_case(in_string):
     return ''.join([s.capitalize() for s in in_string.split('_')])
 
 def get_metric_providers(config):
-    metric_providers_keys = config['measurement']['metric-providers'].keys()
+    architecture = get_architecture()
+    if 'common' in config['measurement']['metric-providers']:
```
Comment (ArneTR) [inline lib/utils.py:35]: This function will fail if the "common" key does not contain any active providers.

File: runner.py [hunk 3]
```diff
@@ -178,17 +175,25 @@ def import_metric_providers(self):
         config = GlobalConfig().config
 
         print(TerminalColors.HEADER, '\nImporting metric providers', TerminalColors.ENDC)
+
+        # if there's a more elegant way to do this, I don't know it
+        if 'common' in config['measurement']['metric-providers']:
```
Comment (ArneTR) [inline runner.py:180]: Why don't you use the get_metric_providers function here?

As I see it has to be reduced a bit and the splitting to be done manually afterwards, but then you could reuse that part of the code

**Review by ArneTR — CHANGES_REQUESTED** [review]

File: metric_providers/lm_sensors/abstract_provider.py (hunk 1 above)
Comment (dan-mm) [reply metric_providers/lm_sensors/abstract_provider.py]: changed to ['linux']

File: lib/utils.py (hunk 2 above)
Comment (dan-mm) [reply lib/utils.py:35]: good catch, pushed a check against this (and the same was true if there were no providers under the architecture key as well)

File: runner.py (hunk 3 above)
Comment (dan-mm) [reply runner.py]: good call, did the refactor and pushed

**Pull request merged**